tests/test_pipelines_common.py

```diff
@@ -9,4 +27,347 @@ class PipelineTesterMixin:
     equivalence of dict and tuple outputs, etc.
     """
 
-    pass
+    # set these parameters to False in the child class if the pipeline does not support the corresponding functionality
+    test_attention_slicing = True
+    test_cpu_offload = True
+    test_xformers_attention = True
+
+    @property
+    def pipeline_class(self) -> Union[Callable, DiffusionPipeline]:
+        raise NotImplementedError(
+            "You need to set the attribute `pipeline_class = ClassNameOfPipeline` in the child test class. "
+            "See existing pipeline tests for reference."
+        )
+
+    def get_dummy_components(self):
+        raise NotImplementedError(
+            "You need to implement `get_dummy_components(self)` in the child test class. "
+            "See existing pipeline tests for reference."
+        )
+
+    def get_dummy_inputs(self, device, seed=0):
+        raise NotImplementedError(
+            "You need to implement `get_dummy_inputs(self, device, seed)` in the child test class. "
+            "See existing pipeline tests for reference."
+        )
+
+    def tearDown(self):
+        # clean up the VRAM after each test in case of CUDA runtime errors
+        super().tearDown()
+        gc.collect()
+        torch.cuda.empty_cache()
+
+    def test_save_load_local(self):
+        if torch_device == "mps" and self.pipeline_class in (
+            DanceDiffusionPipeline,
+            CycleDiffusionPipeline,
+            StableDiffusionImg2ImgPipeline,
+        ):
+            # FIXME: inconsistent outputs on MPS
+            return
```

Comment:
cc @pcuenca I might need your expertise with these, whenever you have time :sweat_smile:

Reply:
As discussed offline let's maybe open an issue here. But it's very nice that we spotted these bugs :-)